src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagCache.cs

```diff
@@ -0,0 +1,64 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class CustomTagCache : ICustomTagCache
+    {
+        private CustomTagList _customtags;
+        private ICustomTagStore _customTagStore;
+        private TimeSpan _expirationInterval;
+        private DateTimeOffset? _lastRefreshTime;
+
+        public CustomTagCache(ICustomTagStore customTagStore, TimeSpan expirationInterval)
+        {
+            EnsureArg.IsNotNull(customTagStore);
+            _customtags = null;
+            _lastRefreshTime = null;
+            _customTagStore = customTagStore;
+            _expirationInterval = expirationInterval;
+        }
+
+        public DateTimeOffset? LastRefreshTime { get => _lastRefreshTime; }
+
+        public async Task<CustomTagList> GetCustomTagsAsync(CancellationToken cancellationToken = default)
```

Comment:
I think this should always require a cancellation token - there shouldn't be a default value.

Reply:
I'm not very sure if that's true for all case, I just followed how other interface does. :) 
e.g: https://github.com/microsoft/dicom-server/blob/master/src/Microsoft.Health.Dicom.Core/Features/ChangeFeed/IChangeFeedService.cs